daemon/cdi.go

```diff
@@ -19,15 +20,28 @@ type cdiHandler struct {
 	registry *cdi.Cache
 }
 
-// RegisterCDIDriver registers the CDI device driver.
+// registerCDIDriver registers the CDI device driver.
 // The driver injects CDI devices into an incoming OCI spec and is called for DeviceRequests associated with CDI devices.
 // If the list of CDI spec directories is empty, the driver is not registered.
-func RegisterCDIDriver(cdiSpecDirs ...string) {
+func registerCDIDriver(cdiSpecDirs ...string) {
 	driver := newCDIDeviceDriver(cdiSpecDirs...)
 
 	registerDeviceDriver("cdi", driver)
 }
 
+// CDIEnabled returns true if CDI feature wasn't explicitly disabled via
+// features.
+func CDIEnabled(conf *config.Config) bool {
+	if runtime.GOOS != "linux" {
+		return false
+	}
+	val, ok := conf.Features["cdi"]
+	if !ok {
+		return true
+	}
+	return val
+}
+
```

Comment:
Could we keep this in cmd/dockerd and in daemon just check for `len(cfg.CDISpecDirs) > 0`?